### PR TITLE
Use named casts on openrct2/rct*

### PR DESCRIPTION
**Pull request by tupaschoal**
Helps #11158

### CODE REVIEW

File: src/openrct2/rct1/S4Importer.cpp [hunk 1]
```diff
@@ -1812,7 +1813,7 @@ class S4Importer final : public IParkImporter
             if (_s4.marketing_status[i] & CAMPAIGN_ACTIVE_FLAG)
             {
                 MarketingCampaign campaign;
-                campaign.Type = (uint8_t)i;
+                campaign.Type = static_cast<uint8_t>(i);
```
Comment (Gymnasiast) [inline src/openrct2/rct1/S4Importer.cpp]: Perhaps it's better to just change the loop variable to a `uint8_t`.

File: src/openrct2/rct12/RCT12.cpp [hunk 2]
```diff
@@ -483,7 +483,7 @@ bool is_user_string_id(rct_string_id stringId)
 
 std::string RCT12::RemoveFormatCodes(const std::string_view& s)
 {
-    constexpr auto RCT12_MULTIBYTE_PREFIX = (char)(uint8_t)0xFF;
+    constexpr auto RCT12_MULTIBYTE_PREFIX = static_cast<char>(static_cast<uint8_t>(0xFF));
```
Comment (Gymnasiast) [inline src/openrct2/rct12/RCT12.cpp]:
```suggestion
    constexpr char RCT12_MULTIBYTE_PREFIX = 0xFF;
```
?

File: src/openrct2/rct12/RCT12.h [hunk 3]
```diff
@@ -276,6 +276,7 @@ struct RCT12TileElement : public RCT12TileElementBase
     uint8_t pad_04[4];
     template<typename TType, RCT12TileElementType TClass> TType* as() const
     {
+        // TODO: CAST-IMPROVEMENT-NEEDED
         return (RCT12TileElementType)GetType() == TClass ? (TType*)this : nullptr;
```
Comment (duncanspumpkin) [inline src/openrct2/rct12/RCT12.h]: static_cast<RCT12TileElementType>(GetType()) == TClass ? reinterpret_cast<TType*>(this) : nullptr;

Comment (Gymnasiast) [reply src/openrct2/rct12/RCT12.h]: Perhaps GetType should just return `RCT12TileElementType`, so we don't need a cast for that.

File: src/openrct2/rct12/RCT12.cpp (hunk 2 above)
Comment (tupaschoal) [reply src/openrct2/rct12/RCT12.cpp:486]: I'm doing this update in batches, so I don't pay too much attention on what could actually be simpler, because it's easier to go after named casts later, than C-style casts.

tl;dr Doing your change yields a warning that becomes an error

Comment (ZehMatt) [reply src/openrct2/rct12/RCT12.cpp:486]: Because 0xFF is not of type char, '\xFF' would work.

File: src/openrct2/rct2/T6Importer.cpp [hunk 4]
```diff
@@ -173,7 +173,9 @@ class TD6Importer final : public ITrackImporter
                 _stream.SetPosition(_stream.GetPosition() - 1);
                 _stream.Read(&t6EntranceElement, sizeof(rct_td6_entrance_element));
                 TrackDesignEntranceElement entranceElement{};
-                entranceElement.z = (t6EntranceElement.z == (int8_t)(uint8_t)0x80) ? -1 : t6EntranceElement.z;
+                entranceElement.z = (t6EntranceElement.z == static_cast<int8_t>(static_cast<uint8_t>(0x80)))
+                    ? -1
+                    : t6EntranceElement.z;
```
Comment (Gymnasiast) [inline src/openrct2/rct2/T6Importer.cpp:178]: There **has** to be a better way. This is so ugly.

File: src/openrct2/rct2/S6Importer.cpp [hunk 5]
```diff
@@ -871,7 +871,7 @@ class S6Importer final : public IParkImporter
         {
             int32_t quadIndex = sceneryEntryIndex >> 5;
             int32_t bitIndex = sceneryEntryIndex & 0x1F;
-            bool invented = (_s6.researched_scenery_items[quadIndex] & ((uint32_t)1 << bitIndex));
+            bool invented = (_s6.researched_scenery_items[quadIndex] & (static_cast<uint32_t>(1) << bitIndex));
```
Comment (Gymnasiast) [inline src/openrct2/rct2/S6Importer.cpp:874]: `1ULL << bitIndex` would be a heck of a lot clearer and less ugly.

File: src/openrct2/rct12/RCT12.cpp [hunk 6]
```diff
@@ -335,7 +335,7 @@ colour_t RCT12SmallSceneryElement::GetSecondaryColour() const
 
 bool RCT12SmallSceneryElement::NeedsSupports() const
 {
-    return (bool)(colour_1 & MAP_ELEM_SMALL_SCENERY_COLOUR_FLAG_NEEDS_SUPPORTS);
+    return static_cast<bool>(colour_1 & MAP_ELEM_SMALL_SCENERY_COLOUR_FLAG_NEEDS_SUPPORTS);
```
Comment (Gymnasiast) [inline src/openrct2/rct12/RCT12.cpp:338]:
```suggestion
    return (colour_1 & MAP_ELEM_SMALL_SCENERY_COLOUR_FLAG_NEEDS_SUPPORTS) != 0;
```

Comment (duncanspumpkin) [reply src/openrct2/rct12/RCT12.cpp:338]: I don't think you even need that it should decay fine to a bool without anything elese.

**Comment by tupaschoal** [comment]
Issues are dealt with

File: src/openrct2/rct2/S6Importer.cpp [hunk 7]
```diff
@@ -841,7 +841,7 @@ class S6Importer final : public IParkImporter
         {
             int32_t quadIndex = rideType >> 5;
             int32_t bitIndex = rideType & 0x1F;
-            bool invented = (_s6.researched_ride_types[quadIndex] & ((uint32_t)1 << bitIndex));
+            bool invented = (_s6.researched_ride_types[quadIndex] & (static_cast<uint32_t>(1) << bitIndex));
```
Comment (Gymnasiast) [inline src/openrct2/rct2/S6Importer.cpp:844]: I probably should have been clearer, but I would like you to replace all occurrences of `static_cast<uint32_t>(1)` with `1ULL` in this file (not just the one on line 874).

**Comment by tupaschoal** [comment]
Found 3 other places to be simplified with `1UL`. As soon as CI is green I'll squash
